### DISCUSSION TITLE
Copy/paste is not working (host Win10, guest Debian bookworm)

### DISCUSSION THREAD

**Discussion by paulvirtuel** [neutrinolabs/xrdp#2856] [Q&A]
I am using win10 remote desktop to connect to a Hyper-V Linux Debian Bookworm virtual machine where I installed xrdp.

So far I am not sure what is going on but I remember being able to copy/paste both ways from/to win10 and the debian vm but now it seems like it works only from time to time and not always in both directions.

Not sure what to do to fix this or how to debug this intermittent problem.

**Comment by matt335672** [comment]
Possible problem areas for the Bookworm version of xrdp (0.9.21.1) :-
1) chansrv is crashing for some currently unidentified reason.
2) xrdp is not connecting to chansrv when reconnecting to the session. This can be because you already have a client connection active - at the moment xrdp doesn't handle this well. See also [this wiki page](https://github.com/neutrinolabs/xrdp/wiki/Copy-Paste-and-network-drives-don't-work-on-a-reconnect)
3) Large pastes can fail (#1839)
4) Unicode characters not in the BMP can cause a paste to fail (#2603)
5) Trying to paste a string when an image is on the clipboard (#2763)

**Reply by paulvirtuel** [reply]
Thanks for your help.  In my case I don't think 3, 4 and 5 are related to the problem I am having.  I have tried the kill old xrdp script and it often helps but not all the time.  
Keep up the good work.  It is a great tool.
